docs/rfcs/2022-12-09-ephemeral-conversations.md

```diff
@@ -0,0 +1,208 @@
+
+\***
+
+APIs:
+
+- `APIStartEC ContactId` - sends `XECInv`, in UI ephemeral chat view is opened, disabled/progress indicator until ephemeral conversation is negotiated.
+- `APIJoinEC ContactId` - sends `XECAcpt`, in UI ephemeral chat view is opened, disabled/progress indicator until ephemeral conversation is negotiated.
+- `APIOpenEC ContactId` - loads chat items (?) for current ephemeral conversation, opens ephemeral chat view.
+- api to reject? or just allow to delete chat item?
+- `APIEndEC ContactId` - sends `XECEnd`, deletes connection, ephemeral conversation entity and chat items, removes from `currentECs` state, deletes `CIECInvitation` chat item, in UI chat is closed.
+- terminal counterparts
+
+ChatResponses (mirroring chat item updates for terminal):
+
+- `CRECInvitationSent {contact :: Contact}`
+- `CRECInvitationReceived {contact :: Contact}`
+- `CRECAccepted {contact :: Contact}`
+- `CRECAcceptReceived {contact :: Contact}`
+- `CRECEnded {contact :: Contact}`
+
+ChatErrors:
+
+- `CEECNegotiationError {contactId :: ContactId}` - both sent `XECInv`, failed to establish connection, etc.
+- `CENoCurrentEC` - on trying to open, accept, end.
+- `CEECState {currentECState :: ECStateTag}` - on state errors
+
+\***
+
+Chat item content:
+
+``` haskell
+data CIContent (d :: MsgDirection) where
+  ...
+  CIRcvECInvitation ECStateTag -> CIContent 'MDRcv
+  CISndECInvitation ECStateTag -> CIContent 'MDSnd
+  ...
+```
+
+is there a need for more detailed `CIECStatus` or state tag will suffice? (see CICallStatus)
+
+\***
+
+New chat type, direction, chat info:
+
+``` haskell
+data ChatType = ... | CTEphemeral ...
+
+-- new ChatType requires processing cases on ChatRef in all APIs based on it, which may be a good thing
+-- e.g. automatically requires separate APISendMessage api
+
+data ChatInfo (c :: ChatType) where
+  ...
+  EphemeralChat :: ChatInfo 'CTEphemeral -- no additional information required?
+  ...
+
+data CIDirection (c :: ChatType) (d :: MsgDirection) where
+  ...
+  CIEphemeralSnd :: CIDirection 'CTEphemeral 'MDSnd
+  CIEphemeralRcv :: CIDirection 'CTEphemeral 'MDRcv
+  ...
+
+-- same for `data CIQDirection (c :: ChatType)`
+
+-- same for `data SChatType (c :: ChatType)`
+```
+
+Maybe it should be a new dimension and not ChatType, though it may have to be drastically different for groups and easier expressed as a separate `CTEphemeralGroup` ChatType.
+
+Pros of not having it as contact's flag/dimension:
+
+- no special casing in loading chat previews
+- no special casing in APIs, instead it's fully fledged ChatType
+- separate table for entity - cleaner deletion, no gaps
+- can have separate ConnectionEntity, though it may be a con
+
+\***
+
+New ConnectionEntity?
+
+- `RcvDirectEphemeralMsgConnection {entityConnection :: Connection}`
+
+can allow to easily prohibit many protocol messages, e.g. calls, groups, etc.
+
+\***
+
+Database changes:
+
+```sql
+CREATE TABLE ephemeral_conversations(
```

Comment:
This doesn't seem necessary at all if it's just a flag on the contact, and all the rest of the state is in memory.